src/Uuid.php

```diff
@@ -649,6 +649,9 @@ public static function setFactory(UuidFactoryInterface $factory): void
      *
      * @return UuidInterface A UuidInterface instance created from a binary
      *     string representation
+     *
+     * @psalm-pure note: changing the internal factory is an edge case not covered by purity invariants,
```

Comment:
Important to know: `@psalm-`-prefixed annotations are ignored by current (known) annotation scanner libraries